evm/src/cpu/simple_logic/not.rs

```diff
@@ -0,0 +1,65 @@
+use plonky2::field::extension_field::Extendable;
+use plonky2::field::field_types::Field;
+use plonky2::field::packed_field::PackedField;
+use plonky2::hash::hash_types::RichField;
+use plonky2::iop::ext_target::ExtensionTarget;
+
+use crate::constraint_consumer::{ConstraintConsumer, RecursiveConstraintConsumer};
+use crate::cpu::columns;
+
+const LIMB_SIZE: usize = 16;
+const ALL_1_LIMB: u64 = (1 << LIMB_SIZE) - 1;
+
+pub fn generate<F: RichField>(lv: &mut [F; columns::NUM_CPU_COLUMNS]) {
+    let is_not_filter = lv[columns::IS_NOT].to_canonical_u64();
+    if is_not_filter == 0 {
+        return;
+    }
+    assert_eq!(is_not_filter, 1);
+
+    for (input_col, output_col) in columns::SIMPLE_LOGIC_INPUT0.zip(columns::SIMPLE_LOGIC_OUTPUT) {
+        let input = lv[input_col].to_canonical_u64();
+        assert_eq!(input >> LIMB_SIZE, 0);
+        let output = input ^ ALL_1_LIMB;
```

Comment:
nit: `!input` seems a bit more direct

Reply:
That inverts all the bits though, whereas I want to invert the bottom `LIMB_SIZE` bits.

Reply:
Oh right!